email-verify.php

```diff
@@ -246,10 +250,19 @@ public function authorize_page_id(){
 	}
 
 	/**
-	 * Does user needs email validation?
+	 * Return false if user doesn't need validation, stored hash if it does
 	 */
 	public function needs_validation( $user_id ){
-		return boolval( get_user_meta( $user_id, 'verify-lock', true ) );
+		$needs_verification = false;
+		// If the verification metadata doesn't even exist, assume token hasn't been issued. That would indicate that
+		// the user registered at a time when this plugin was not installed/active. Don't require verification in that case.
+		if( metadata_exists( 'user', $user_id, "verify-lock" ) ) {
```

Comment:
If this user registered before the plugin was installed, `metadata_exists()` will return false here, which makes the whole function return false, which means they don't need validation.